cedar-policy-core/src/extensions.rs

```diff
@@ -233,7 +293,7 @@ pub mod extension_function_lookup_errors {
     // Don't make fields `pub`, don't make breaking changes, and use caution
     // when adding public methods.
     #[derive(Debug, PartialEq, Eq, Clone, Error)]
-    #[error("extension function `{name}` does not exist")]
+    #[error("internal invariant violated. Extension function `{name}` does not exist")]
```

Comment:
Might be worth commenting why this is an internal invariant violation, and why you won't get this error from a policy that calls `foo()` (in either JSON or Cedar format), or entity JSON data that calls `foo()`

Reply:
Looked at this closer, and the supposed invariant is violated when parsing entity data:

```json
    {
        "uid": {  "type":"User", "id":"bob" },
        "attrs": {
            "score": { "__extn": { "fn": "foo", "arg": "0.899" }}
        },
        "parents": []
    }
```
```
  × failed to parse entities from file test.json
  ├─▶ error during entity deserialization
  ╰─▶ failed to evaluate attribute `score` of `User::"bob"`: internal invariant violated. Extension function `foo` does not exist
```

Not a Cedar bug because we're getting an error exactly where we want one. Just means this error variant can occur legitimately without some more refactoring

Reply:
IMO not worth restructuring just so we can say this is an internal invariant violation.

Reply:
Makes sense to me